src/zope/paste/tests.py

```diff
@@ -128,7 +128,7 @@ def test_serving_test_app():
     >>> from zope.paste import serve
     >>> server = threading.Thread(target=serve.serve, args=(['app.ini'],))
     >>> server.start()
-    >>> import time; time.sleep(5)  # doctest: +ELLIPSIS
+    >>> import time; time.sleep(10)  # doctest: +ELLIPSIS
     Serving on http://localhost:...
```

Comment:
What if the thread is faster than expected and prints the `Serving on ...` right after `server.start()`?

Ideally we'd have a synchronization primitive of some kind and could do `>>> server.start(); wait_for_server_to_be_ready()`.

Reply:
Ok, that's not what happens.  I'm not sure what happens -- waitress stopped printing 'Serving on ...' by default?

Reply:
It seems like `waitress >= 2` needs a custom logging setup to get it printed here.

Reply:
See https://github.com/zopefoundation/zope.paste/pull/3/commits/967f784c5aa40fa1fed3eea07a203fd829a84e5e.